mo_net/samples/mnist.py

```
@@ -0,0 +1,186 @@
+        if output_paths.peek() is None:
+            logger.error(
+                "No model file found in the output directory and no model path provided."
+            )
+            sys.exit(1)
+        model_path = max(output_paths, key=lambda p: p.stat().st_mtime)
+        logger.info(f"Using latest model file: {model_path}")
+    if not model_path.exists():
+        logger.error(f"File not found: {model_path}")
+        sys.exit(1)
+
+    model = Model.load(open(model_path, "rb"))
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)